lightning/src/onion_message/messenger.rs

```diff
@@ -209,9 +209,17 @@ impl<Signer: Sign, K: Deref, L: Deref> OnionMessenger<Signer, K, L>
 			msg.onion_routing_packet.hmac, control_tlvs_ss)
 		{
 			Ok((Payload::Receive {
-				control_tlvs: ReceiveControlTlvs::Unblinded(ReceiveTlvs { path_id })
+				control_tlvs: ReceiveControlTlvs::Unblinded(ReceiveTlvs { path_id }), reply_path,
 			}, None)) => {
-				log_info!(self.logger, "Received an onion message with path_id: {:02x?}", path_id);
+				log_info!(self.logger,
+					"Received an onion message with path_id: {:02x?} and {}reply_path",
+						path_id, if reply_path.is_some() { "" } else { "no " });
+				#[cfg(test)]
+				{
+					if let Some(reply_path) = reply_path {
+						self.send_onion_message(&[], Destination::BlindedRoute(reply_path), None).unwrap();
+					}
+				}
```

Comment:
Hmmm... could we just put this in the relevant tests?

Reply:
I'm not sure how, unless we wanna add some test-only struct field handler for reply paths instead?

Reply:
To take a step back, I guess it was unclear to me what new behavior we're trying to test. Looks like it's primarily that `send_onion_message` will *encode a reply path* for the destination. Would it be sufficient to simply update the logging assertion in `pass_along_path` to check if a reply path was given?

Presumably, after handling is added, both the test-only code and logging assertions would no longer be necessary as handling could be tested properly. Otherwise, seems like we're adding test-only behavior to test a behavior that doesn't yet exist (i.e., *using a reply path* when handling).

Reply:
>  Looks like it's primarily that send_onion_message will encode a reply path for the destination.

FWIW, my thinking was that we're not just testing this^, but also that we can actually use a reply path that we decode, and make sure that reply reaches its destination (so just feature correctness basically). 

I don't feel too strongly though, since it's fair to say that it'll be tested later anyway. Thoughts?

Reply:
Isn't that testing the test rather than the system under test? (i.e., testing that the correct reply path was passed to `send_onion_message`)

Reply:
I mean (a version of) this code will be not-`#[cfg(test)]` eventually. But it is fair that the behavior doesn't exist now, I'll remove this + the relevant tests